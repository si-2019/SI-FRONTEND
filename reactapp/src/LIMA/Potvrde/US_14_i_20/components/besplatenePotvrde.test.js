import React from "react";
import BesplatnePotvrde from "./besplatnePotvrde";
import { shallow } from "enzyme";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

it("Komponenta svrha potvrde se ispravno renda", () => {
  const komponenta = shallow(<BesplatnePotvrde />);
  expect(komponenta).toMatchSnapshot();
});
it("U komponenti se nalazi broj potvrda", () => {
  const komponenta = shallow(<BesplatnePotvrde />);
  expect(komponenta.find("#broj_potvrda").exists()).toBe(true);
});
