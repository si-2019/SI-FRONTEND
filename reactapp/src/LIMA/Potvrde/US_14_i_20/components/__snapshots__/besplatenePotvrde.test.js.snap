// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Komponenta svrha potvrde se ispravno renda 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BesplatnePotvrde />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="col"
        >
          <p>
            Broj bespaltnih potvrda: 
          </p>
        </div>,
        <div
          className="col"
        >
          <span
            className="badge-pill badge-primary"
            id="broj_potvrda"
          >
            5
          </span>
        </div>,
      ],
      "className": "row",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <p>
            Broj bespaltnih potvrda: 
          </p>,
          "className": "col",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Broj bespaltnih potvrda: ",
          },
          "ref": null,
          "rendered": "Broj bespaltnih potvrda: ",
          "type": "p",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <span
            className="badge-pill badge-primary"
            id="broj_potvrda"
          >
            5
          </span>,
          "className": "col",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 5,
            "className": "badge-pill badge-primary",
            "id": "broj_potvrda",
          },
          "ref": null,
          "rendered": 5,
          "type": "span",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="col"
          >
            <p>
              Broj bespaltnih potvrda: 
            </p>
          </div>,
          <div
            className="col"
          >
            <span
              className="badge-pill badge-primary"
              id="broj_potvrda"
            >
              5
            </span>
          </div>,
        ],
        "className": "row",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <p>
              Broj bespaltnih potvrda: 
            </p>,
            "className": "col",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Broj bespaltnih potvrda: ",
            },
            "ref": null,
            "rendered": "Broj bespaltnih potvrda: ",
            "type": "p",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span
              className="badge-pill badge-primary"
              id="broj_potvrda"
            >
              5
            </span>,
            "className": "col",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 5,
              "className": "badge-pill badge-primary",
              "id": "broj_potvrda",
            },
            "ref": null,
            "rendered": 5,
            "type": "span",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
